tajo-cli/src/main/java/org/apache/tajo/cli/tsql/TajoCli.java

```diff
@@ -374,11 +387,43 @@ private void initCommands() {
         System.err.println(e.getMessage());
         throw new RuntimeException(e.getMessage());
       }
+
+      // make completers for console auto-completion
+      compList.add(cmd.getArgumentComplementer());
+
       commands.put(cmd.getCommand(), cmd);
       for (String alias : cmd.getAliases()) {
         commands.put(alias, cmd);
       }
     }
+
+    cliCompleter = new AggregateCompleter(compList);
+
+    sqlCompleter = new ArgumentCompleter(
+        new ArgumentCompleter.AbstractArgumentDelimiter() {
+          @Override
+          public boolean isDelimiterChar(CharSequence buf, int pos) {
+            char c = buf.charAt(pos);
+            return Character.isWhitespace(c) || !(Character.isLetterOrDigit(c)) && c != '_';
+          }
+        },
+        new StringsCompleter(getKeywords())
+    );
+  }
+
+  private Collection<String> getKeywords() {
+    List<String> klist = new ArrayList<>();
```

Comment:
It looks like a ambiguous name. How about rename it to ``keywords`` or ``keywordList``?